datasync/chaindatafetcher/kafka/consumer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+)
+
+// Consumer subscribes a topic pushed by a producer in the kafka cluster.
+type Consumer struct {
+	cancelCh chan struct{}
+	handler  map[string]func(*sarama.ConsumerMessage) error
+	members  sarama.ConsumerGroup
+	ctx      context.Context
+	isActive bool
+}
+
+func NewConsumer(ctx context.Context, members sarama.ConsumerGroup) *Consumer {
+	return &Consumer{
+		cancelCh: make(chan struct{}),
+		handler:  map[string]func(*sarama.ConsumerMessage) error{},
+		ctx:      ctx,
+		members:  members,
+		isActive: false,
+	}
+}
+
+func (c *Consumer) Subscribe(topic string, handler func(*sarama.ConsumerMessage) error) error {
+	if c.handler[topic] != nil {
+		return nil
```

Comment:
Why don't you return an error like `already exist topic?

Reply:
The `Subcribe` method should be called only once. If it is called again with the same topic, it means that a goroutine already subscribes the topic. That is why it returns nil instead of error.